src/realm/object-store/object_store.cpp

```diff
@@ -850,19 +850,20 @@ static void apply_post_migration_changes(Group& group, std::vector<SchemaChange>
 void ObjectStore::apply_schema_changes(Transaction& group, uint64_t schema_version, Schema& target_schema,
                                        uint64_t target_schema_version, SchemaMode mode,
                                        std::vector<SchemaChange> const& changes, bool handle_automatically_backlinks,
-                                       std::function<void()> migration_function)
+                                       std::function<void()> migration_function,
+                                       bool set_schema_version_on_version_decrease)
 {
     create_metadata_tables(group);
 
     if (mode == SchemaMode::AdditiveDiscovered || mode == SchemaMode::AdditiveExplicit) {
-        bool target_schema_is_newer =
-            (schema_version < target_schema_version || schema_version == ObjectStore::NotVersioned);
+        bool set_schema = (schema_version < target_schema_version || schema_version == ObjectStore::NotVersioned ||
+                           set_schema_version_on_version_decrease);
```

Comment:
This check should just be deleted entirely as it's long obsolete. It was originally there because we used to the schema version to decide if we should create and remove indexes.